lib/Mojo/TLS.pm

```diff
@@ -0,0 +1,125 @@
+package Mojo::TLS;
+use Mojo::Base -strict;
+
+# TLS support requires IO::Socket::SSL
+use constant TLS => $ENV{MOJO_NO_TLS}
+  ? 0
+  : eval 'use IO::Socket::SSL 1.94 (); 1';
+
+use constant {
+  TLS_READ  => TLS ? IO::Socket::SSL::SSL_WANT_READ()  : 0,
+  TLS_WRITE => TLS ? IO::Socket::SSL::SSL_WANT_WRITE() : 0,
+  TLS_NPN   => TLS ? eval { IO::Socket::SSL->can_npn }        : 0,
+  TLS_ALPN  => TLS ? eval { IO::Socket::SSL->can_alpn }       : 0,
+  TLS_C_SNI => TLS ? eval { IO::Socket::SSL->can_client_sni } : 0,
+  TLS_S_SNI => TLS ? eval { IO::Socket::SSL->can_server_sni } : 0,
+};
+
+use Exporter 'import';
+our @EXPORT_OK = (
+  qw(TLS TLS_READ TLS_WRITE TLS_NPN TLS_ALPN TLS_C_SNI TLS_S_SNI),
+  qw(mojo_protocols selected_protocol)
+);
+
+sub mojo_protocols {
```

Comment:
This is related to APLN. When tls handshake performed client announce list of supported protocols. 'http/1.1' is one of registered by IANA constants https://tools.ietf.org/html/rfc7301#section-6